packages/fx-core/src/component/resource/botService/botRegistration/utils.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IBotRegistration } from "../appStudio/interfaces/IBotRegistration";
+
+export class Utils {
+  public static mergeIBotRegistration(
+    local: IBotRegistration,
+    remote: IBotRegistration
+  ): IBotRegistration {
+    return {
+      botId: local.botId ?? remote.botId,
+      name: local.name ? local.name : remote.name,
```

Comment:
It seems all logic looks like ```local.prop ?? remote.prop```. Why only L12 use this pattern, but the rest use another style?

Reply:
This is because in IBotRegistration, botId is the only one with type `string | undefined` and the rest are `string`.
 `??` wasn't used for the rest, for example, `iconUrl: local.iconUrl ? local.iconUrl : remote.iconUrl`,  because `??` doesn't work for empty strings. 
If `local.iconUrl` is empty, we should respect the value from remote.

Reply:
Can use `local.iconUrl || remote.iconUrl` for simplicity?